lib/content-services/src/lib/search/components/search-input/search-input.component.ts

```diff
@@ -0,0 +1,105 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CommonModule } from '@angular/common';
+import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { TranslateModule } from '@ngx-translate/core';
+
+@Component({
+    selector: 'app-search-input',
+    standalone: true,
+    imports: [CommonModule, MatFormFieldModule, MatInputModule, TranslateModule],
+    templateUrl: `./search-input.component.html`,
+    styleUrls: ['./search-input.component.scss'],
+    encapsulation: ViewEncapsulation.None
+})
+export class SearchInputComponent {
+    @Input()
+    value = '';
+
+    @Input()
+    label = 'SEARCH.INPUT.LABEL';
+
+    @Input()
+    placeholder = 'SEARCH.INPUT.PLACEHOLDER';
+
+    @Input()
+    fields = ['cm:name'];
+
+    @Output()
+    changed = new EventEmitter<string>();
+
+    onSearchInputChanged(event: Event) {
+        const input = event.target as HTMLInputElement;
+        const searchTerm = input.value;
+
+        const query = this.formatSearchQuery(searchTerm, this.fields);
+        if (query) {
+            this.changed.emit(decodeURIComponent(query));
+        }
+    }
+
+    private formatSearchQuery(userInput: string, fields = ['cm:name']): string {
+        if (!userInput) {
+            return null;
+        }
+
+        if (/^http?:\/\//.test(userInput)) {
```

Comment:
@DenysVuika is that wanted change? Earlier you had http[s] now you have http. Intellij suggested using https because it was fine to skip [] because then '?' is used only for 's' letter without need for []. 

If that is fine to keep just http here then fine :)

Reply:
the code is taken from here https://github.com/Alfresco/alfresco-content-app/blob/develop/projects/aca-content/src/lib/components/search/search-results/search-results.component.ts#L222 and I believe it's testing for `http` or `https` ?

Reply:
yes and I think that intellij complains that http[s]? is the same like https? so no need to wrap [s] around s because regex will work in the same way (in both it allows to use http or https). 

But now you skipped s at all in that if, you have just http

Reply:
fixed to `https?` as per suggestion https://stackoverflow.com/a/62948640